python/dlisio/ext/core.cpp

```diff
@@ -605,7 +605,11 @@ PYBIND11_MODULE(core, m) {
             recs.reserve( indices.size() );
             for (auto i : indices) {
                 auto rec = s.at( i );
-                if (rec.isencrypted()) continue;
+                if (rec.isencrypted()){
+                    /* as we don't want to add encrypted data yet, adding
+                       empty record instead to preserve data structure */
+                    rec.data.clear();
```

Comment:
I think this isn't a great fix.

If we see the record is encrypted we should probably be able to return all the bytes intact, and leave dealing with padbytes to whoever's decrypting it. With this, we now have to deal with encrypted records _and_ we destroy the data in the process.

Reply:
I fail to see how it is different from what we had before? I wanted to avoid changing current logic.
Previously if record was encrypted we as well just ignored the record, so it was as is we never added the data. Right now we add placeholder instead.
Do you have a better suggestion for these records?

Reply:
>  I wanted to avoid changing current logic.

Don't, current logic is awful.

> I fail to see how it is different from what we had before?

I suppose it's not too different, except we're now still signaling an error. It's different because if the record is encrypted we don't even try to trim, whereas before we always trimmed. When encrypted, we should probably keep on going, return the byte array, but flag the fact that it's rubbish until decrypted.

> Previously if record was encrypted we as well just ignored the record, so it was as is we never added the data.

Sure, and this is fine. There should be a recovery though, for the cases where the caller _can_ decrypt.

Reply:
Changed for encrypted records not to be treated any differently.
User has to check record.isencrypted() to find out if this record is garbage or not.
If I still misunderstand what you are communicating, lets talk through slack.